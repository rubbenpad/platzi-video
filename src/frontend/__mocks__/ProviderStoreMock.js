import React from 'react'
import { createStore, compose, applyMiddleware } from 'redux'
import thunk from 'redux-thunk'
import { Router } from 'react-router'
import { createBrowserHistory } from 'history'
import { Provider } from 'react-redux'
import reducer from '../reducers'
import initialState from '../initialState'

const store = createStore(
  reducer,
  initialState,
  compose(applyMiddleware(thunk))
)
const history = createBrowserHistory()

const ProviderStoreMock = ({ children }) => (
  <Provider store={store}>
    <Router history={history}>{children}</Router>
  </Provider>
)

export default ProviderStoreMock
