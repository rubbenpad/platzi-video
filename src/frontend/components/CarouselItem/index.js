import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'
import { setUserMovie, deleteUserMovie } from '../../actions'

import {
  Item,
  Cover,
  Detail,
  Action,
  AddButton,
  PlayButton,
  CheckButton,
} from './styles'

const CarouselItem = props => {
  const { isLibrary, id, title, cover, year, contentRating, duration } = props

  const handleSetFavorite = () => {
    return props.setUserMovie({
      movie: {
        id,
        title,
        cover,
        year,
        contentRating,
        duration,
      },
    })
  }

  const handleDeleteFavorite = movieId => {
    return props.deleteUserMovie(movieId)
  }

  return (
    <Item>
      <Cover src={cover} alt="Video Cover" />
      <Detail>
        <Action>
          <Link to={`/player/${id}`}>
            <PlayButton />
          </Link>
          {!isLibrary ? (
            <AddButton onClick={handleSetFavorite} />
          ) : (
            <CheckButton onClick={() => handleDeleteFavorite(id)} />
          )}
        </Action>
        <h3>{title}</h3>
        <p>{`${year} ${contentRating} ${duration}min`}</p>
      </Detail>
    </Item>
  )
}

CarouselItem.propTypes = {
  isLibrary: PropTypes.bool,
  id: PropTypes.string,
  title: PropTypes.string,
  cover: PropTypes.string,
  year: PropTypes.number,
  contentRating: PropTypes.string,
  duration: PropTypes.number,
}

const mapDispatchToProps = {
  setUserMovie,
  deleteUserMovie,
}

export default connect(null, mapDispatchToProps)(CarouselItem)
