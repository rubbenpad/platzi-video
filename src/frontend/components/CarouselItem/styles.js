import styled from 'styled-components';
import { FaPlus, FaPlay, FaCheck } from 'react-icons/fa';

const width = 250;
const height = width / (16 / 9);

export const Item = styled.div`
  width: ${width}px;
  height: ${height}px;
  margin-right: 10px;
  display: inline-block;
  border-radius: 3px;
  position: relative;
  cursor: pointer;
  font-size: 20px;
  overflow: hidden;
  transition: 0.45s all;
  transform-origin: center left;
`;

export const Cover = styled.img`
  width: ${width}px;
  height: ${height}px;
  object-fit: cover;
`;

export const Detail = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  align-items: flex-start;
  padding-bottom: 20px;
  position: absolute;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  opacity: 0;
  background: linear-gradient(
    to top,
    rgba(0, 0, 0, 0.9) 0%,
    rgba(0, 0, 0, 0) 100%
  );
  transition: 0.45s opacity;

  h3 {
    margin: 0 0 0 10px;
    font-size: 12px;
  }

  p {
    margin: 0 0 0 10px;
    font-size: 10px;
    opacity: 0.9;
    color: #fff;
  }
`;

export const Action = styled.div`
  margin-left: 1rem;
`;

export const AddButton = styled(FaPlus)`
  width: 16px;
  margin-right: 5px;
  color: #fff;
  transition: opacity 0.5s ease;

  &:hover {
    transform: scale(1.1);
    opacity: 0.9;
  }
`;

export const PlayButton = styled(FaPlay)`
  width: 16px;
  margin-right: 8px;
  color: #fff;
  transition: opacity 0.5s ease;

  &:hover {
    transform: scale(1.1);
    opacity: 0.9;
  }
`;

export const CheckButton = styled(FaCheck)`
  width: 16px;
  margin-right: 8px;
  color: #fff;
  transition: opacity 0.5s ease;

  &:hover {
    transform: scale(1.1);
    opacity: 0.9;
  }
`;
