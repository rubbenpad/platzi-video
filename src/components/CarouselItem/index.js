import React from 'react';
import { connect } from 'react-redux';
import { setFavorite, deleteFavorite } from '../../actions';

import { Item, Cover, Detail, Action } from './styles';
import playIcon from '../../assets/play-icon.png';
import addIcon from '../../assets/plus-icon.svg';
import deleteIcon from '../../assets/delete-icon.svg';

const CarouselItem = (props) => {
  const { isLibrary, id, title, cover, year, contentRating, duration } = props;
  const handleSetFavorite = () => {
    return props.setFavorite({
      id,
      title,
      cover,
      year,
      contentRating,
      duration,
    });
  };

  const handleDeleteFavorite = (itemId) => {
    return props.deleteFavorite(itemId);
  };

  return (
    <Item>
      <Cover src={cover} alt="Video Cover" />
      <Detail>
        <Action>
          <img src={playIcon} alt="Play Button" />
          {!isLibrary ? (
            <img src={addIcon} alt="Add Button" onClick={handleSetFavorite} />
          ) : (
            <img
              src={deleteIcon}
              alt="Delete Button"
              onClick={() => handleDeleteFavorite(id)}
            />
          )}
        </Action>
        <h3>{title}</h3>
        <p>{`${year} ${contentRating} ${duration}min`}</p>
      </Detail>
    </Item>
  );
};

const mapDispatchToProps = {
  setFavorite,
  deleteFavorite,
};

export default connect(null, mapDispatchToProps)(CarouselItem);
