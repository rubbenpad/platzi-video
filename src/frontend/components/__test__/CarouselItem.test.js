import React from 'react'
import { create } from 'react-test-renderer'
import ProviderStoreMock from '../../__mocks__/ProviderStoreMock'
import CarouselItem from '../CarouselItem'

describe('Carousel component test', () => {
  test('Match Snapshot', () => {
    const carouselItem = create(
      <ProviderStoreMock>
        <CarouselItem />
      </ProviderStoreMock>
    )
    expect(carouselItem.toJSON()).toMatchSnapshot()
  })
})
