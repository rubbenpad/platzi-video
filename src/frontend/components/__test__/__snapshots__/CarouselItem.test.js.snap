// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel component test Match Snapshot 1`] = `
<div
  className="sc-bdVaJa cFlflQ"
>
  <img
    alt="Video Cover"
    className="sc-bwzfXH gelAAY"
  />
  <div
    className="sc-htpNat gUusgJ"
  >
    <div
      className="sc-bxivhb gfqwgb"
    >
      <a
        href="/player/undefined"
        onClick={[Function]}
      >
        <svg
          className="sc-EHOje iIgAP"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          strokeWidth="0"
          style={
            Object {
              "color": undefined,
            }
          }
          viewBox="0 0 448 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M424.4 214.7L72.4 6.6C43.8-10.3 0 6.1 0 47.9V464c0 37.5 40.7 60.1 72.4 41.3l352-208c31.4-18.5 31.5-64.1 0-82.6z"
          />
        </svg>
      </a>
      <svg
        className="sc-ifAKCX bOXfnb"
        fill="currentColor"
        height="1em"
        onClick={[Function]}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 448 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
        />
      </svg>
    </div>
    <h3 />
    <p>
      undefined undefined undefinedmin
    </p>
  </div>
</div>
`;
